tools/clang/lib/SPIRV/SPIRVEmitter.cpp

```diff
@@ -3130,6 +3141,7 @@ bool SPIRVEmitter::emitEntryFunctionWrapper(const FunctionDecl *decl,
   // Construct the wrapper function signature.
   const uint32_t voidType = theBuilder.getVoidType();
   const uint32_t funcType = theBuilder.getFunctionType(voidType, {});
+  entryFunctionDecl = const_cast<FunctionDecl*>(decl);
```

Comment:
I agree... It's wasteful to keep a class field solely for it. I'll handle it where it needs to be handled. But I will keep the `AddExecutionModeForEntryPoint()` function as it used to be, because I foresee there's going to be a bunch of logic in there to decide the execution mode for different shader stages